tests/smoke/support/page/files/allFiles.ts

```diff
@@ -187,4 +213,132 @@ export class AllFilesPage {
     })
     await page.goto(startUrl)
   }
+
+  async resourceExist({ name }: { name: string }): Promise<boolean> {
+    const { page } = this.actor
+    const folderPaths = name.split('/')
+    const resouceName = folderPaths.pop()
+
+    if (folderPaths.length) {
+      await cta.files.navigateToFolder({ page: page, path: folderPaths.join('/') })
+    }
+
+    const resourceExists = await cta.files.resourceExists({
+      page: page,
+      name: resouceName
+    })
+
+    return resourceExists
+  }
+
+  async numberOfVersions({ resource }: { resource: string }): Promise<number> {
+    const { page } = this.actor
+    const folderPaths = resource.split('/')
+    const resouceName = folderPaths.pop()
+
+    if (folderPaths.length) {
+      await cta.files.navigateToFolder({ page: page, path: folderPaths.join('/') })
+    }
+
+    await cta.files.sidebar.open({ page: page, resource: resouceName })
+    await cta.files.sidebar.openPanel({ page: page, name: 'versions' })
+
+    const elements = await page.$$('//*[@id="oc-file-versions-sidebar"]/table/tbody/tr')
+    return elements.length
+  }
+
+  async deleteResourses({ resource }: { resource: string }): Promise<void> {
```

Comment:
same as numberOfVersions

Reply:
👆

Reply:
fine for me, can you please keep track of it :) Thanks